runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -462,42 +446,89 @@ public void run() {
     LOG.debug("maxWorkItemCommitBytes: {}", maxWorkItemCommitBytes);
   }
 
-  private static WindmillServerStub createWindmillServerStub(
+  public static StreamingDataflowWorker fromOptions(StreamingDataflowWorkerOptions options) {
+    ConcurrentMap<String, ComputationState> computationMap = new ConcurrentHashMap<>();
+    long clientId = clientIdGenerator.nextLong();
+    return new StreamingDataflowWorker(
+        createWindmillServerStub(
+            options,
+            clientId,
+            new WorkHeartbeatResponseProcessor(
```

Comment:
I don't see where this is setup if the forTesting method is used.

Are you sure this isn't the cause of the test failure?

Reply:
it gets set up here in FakeWindmillServer
![Uploading image.png…]()

Reply:
it gets set up here in FakeWindmillServer
![image](https://github.com/apache/beam/assets/29557027/6701f28a-37d6-4a95-aa3d-6632e0d05fb9)

Reply:
figured it out looks like the work isn't being populated in the computation map will fix it

thanks!